packages/date/src/index.js

```diff
@@ -71,8 +71,10 @@ export function setSettings( dateSettings ) {
 			y: 'a year',
 			yy: '%d years',
 		},
+		week: {
+			dow: dateSettings.l10n.start_of_week,
+		},
 	} );
-	momentLib.locale( currentLocale );
```

Comment:
I wonder if we should keep this and apply it in `wp.data.moment` instead. I think the idea of this is that the module is trying to avoid messing with any external global configuration applied to moment. 

Also, it might be good to see how we can get rid of moment entirely from the external API of this module (but this is out of scope of this PR)

Reply:
Thanks @youknowriad, I understand now. So an extension can simply switch or update the locale information on its own.

```js
// Switch to the previously loaded locale
moment.locale( wp.date.getSettings().l10n.locale );

// Or even update the data if need be
moment.updateLocale(  wp.date.getSettings().l10n.locale, { ... }  );
```
This satisfies the needs of our app. I'll close this PR, thanks for you help.

>it might be good to see how we can get rid of moment entirely from the external API

I can agree here. From the perspective of someone consuming the API, its causing more confusion than help. In fact, even switching to the previously loaded locale causes issues, e.g. this breaks usage of moment's format api, `moment().format( 'LL' )` because these formats in `dateSettings`are PHP based.

```js
longDateFormat: {
    LT: dateSettings.formats.time,
    LTS: null,
    L: null,
    LL: dateSettings.formats.date,
    LLL: dateSettings.formats.datetime,
    LLLL: null,
}
```